relayer/chains/evm/listener.go

```diff
@@ -82,6 +82,7 @@ func (p *Provider) Listener(ctx context.Context, lastSavedHeight uint64, blockIn
 					end = latestHeight
 				}
 				blockReqs = append(blockReqs, &blockReq{start: i, end: end, retry: maxBlockQueryFailedRetry})
+				i = end + 1
```

Comment:
I think this is should generate the correct requests: 
```go
start := startHeight
for start <= latestHeight {
    end := i + p.cfg.BlockBatchSize-1
    if end > latestHeight {
	    end = latestHeight
    }
    blockReqs = append(blockReqs, &blockReq{start: start, end: end, retry: maxBlockQueryFailedRetry})
    start = end + 1
}
```

Reply:
we have more elegant solution now without tracking anything.